### PR TITLE
Implement interactive prompt

### PR DESCRIPTION
**Pull request by droctothorpe**
This PR adds an interactive prompt for `app.kubernetes.io/instance` label value selection. It relies on the excellent [survey](https://github.com/AlecAivazis/survey) package. 

It's triggered by the `--prompt` or `-p` flag.

Here's a gif that illustrates usage:

(`grm` is an alias for `go run main.go`). 

Relevant issue: https://github.com/stern/stern/issues/120

### CODE REVIEW

**Comment by droctothorpe** [comment]
Pulled in the latest changes from master and resolved the one merge conflict.

**Comment by superbrothers** [comment]
Could you please update the cli flags section and add an example for new `--prompt` flag on README.md?

**Comment by droctothorpe** [comment]
> Could you please update the cli flags section and add an example for new `--prompt` flag on README.md?

Done. Happy to include a gif or sample output for the prompt as well if you think that makes sense.

**Comment by droctothorpe** [comment]
Addressed the new merge conflict 😭 . Should be gtg.

**Comment by droctothorpe** [comment]
Checking in, @superbrothers.

File: README.md [hunk 1]
```diff
@@ -92,7 +93,7 @@ stern supports outputting custom log messages.  There are a few predefined
 templates which you can use by specifying the `--output` flag:
 
 | output    | description                                                                                           |
-|-----------|-------------------------------------------------------------------------------------------------------|
+| --------- | ----------------------------------------------------------------------------------------------------- |
```
Comment (superbrothers) [inline README.md:96]: Do not include any style changes that are not directly related to the function.

File: README.md [hunk 2]
```diff
@@ -102,7 +103,7 @@ compiled to a Go template and then used for every log message. This Go template
 will receive the following struct:
 
 | property        | type   | description                                 |
-|-----------------|--------|---------------------------------------------|
+| --------------- | ------ | ------------------------------------------- |
```
Comment (superbrothers) [inline README.md:106]: ditto.

File: README.md [hunk 3]
```diff
@@ -113,7 +114,7 @@ The following functions are available within the template (besides the [builtin
 functions](https://golang.org/pkg/text/template/#hdr-Functions)):
 
 | func    | arguments             | description                                                     |
-|---------|-----------------------|-----------------------------------------------------------------|
+| ------- | --------------------- | --------------------------------------------------------------- |
```
Comment (superbrothers) [inline README.md:117]: ditto.

File: cmd/cli.go [hunk 4]
```diff
@@ -127,25 +130,34 @@ func Run() {
 	cmd.RunE = func(cmd *cobra.Command, args []string) error {
 		if opts.version {
 			fmt.Println(buildVersion(version, commit, date))
+
```
Comment (superbrothers) [inline cmd/cli.go:133]: ditto.

File: cmd/cli.go [hunk 5]
```diff
@@ -127,25 +130,34 @@ func Run() {
...
 		if opts.completion != "" {
 			return runCompletion(opts.completion, cmd)
 		}
 
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
 		narg := len(args)
-		if (narg > 1) || (narg == 0 && opts.selector == "" && opts.fieldSelector == "") {
+
```
Comment (superbrothers) [inline cmd/cli.go:145]: ditto.

File: cmd/cli.go [hunk 6]
```diff
@@ -127,25 +130,34 @@ func Run() {
...
+		if (narg > 1) || (narg == 0 && opts.selector == "" && opts.fieldSelector == "") && !opts.prompt {
 			return cmd.Help()
 		}
+
```
Comment (superbrothers) [inline cmd/cli.go:149]: ditto.

File: cmd/cli.go [hunk 7]
```diff
@@ -371,3 +383,64 @@ func makeUnique(items []string) []string {
...
+func selectPods(pods []string) (string, error) {
+	arrow := survey.WithIcons(func(icons *survey.IconSet) {
+		icons.Question.Text = "❯"
+		icons.SelectFocus.Text = "❯"
+		icons.Question.Format = "blue"
+		icons.SelectFocus.Format = "blue"
+	})
+
+	prompt := &survey.Select{
+		Message: "Select label value:",
```
Comment (superbrothers) [inline cmd/cli.go:435]:
```suggestion
		Message: "Select \"app.kubernetes.io/instance\" label value:",
```

File: cmd/cli.go [hunk 8]
```diff
@@ -109,6 +111,7 @@ func Run() {
 	cmd.Flags().StringVar(&opts.completion, "completion", opts.completion, "Outputs stern command-line completion code for the specified shell. Can be 'bash' or 'zsh'")
 	cmd.Flags().StringVar(&opts.template, "template", opts.template, "Template to use for log lines, leave empty to use --output flag")
 	cmd.Flags().StringVarP(&opts.output, "output", "o", opts.output, "Specify predefined template. Currently support: [default, raw, json]")
+	cmd.Flags().BoolVarP(&opts.prompt, "prompt", "p", opts.prompt, "Toggle interactive label selection prompt")
```
Comment (superbrothers) [inline cmd/cli.go:114]: Please mention using `app.kubernetes.io/instance` label.

File: README.md [hunk 9]
```diff
@@ -193,6 +194,12 @@ Output using a custom template with stern-provided colors:
 stern --template '{{.Message}} ({{.Namespace}}/{{color .PodColor .PodName}}/{{color .ContainerColor .ContainerName}}){{"\n"}}' backend
 ```
 
+Trigger the interactive prompt:
```
Comment (superbrothers) [inline README.md:197]: Please mention using the `app.kubernetes.io/instance` label.

File: README.md (hunk 1 above)
Comment (droctothorpe) [reply README.md:96]: Copy that. That's a VSC plugin that auto-formats on save. I'll fix this.

File: cmd/cli.go [hunk 10]
```diff
@@ -112,8 +114,12 @@ func Run() {
 			return runCompletion(opts.completion, cmd)
 		}
 
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
 		narg := len(args)
-		if (narg > 1) || (narg == 0 && opts.selector == "" && opts.fieldSelector == "") {
+
```
Comment (droctothorpe) [inline cmd/cli.go:121]:
```suggestion
```

**Comment by droctothorpe** [comment]
Thanks for the feedback, @superbrothers. I addressed the notes. 

`update-readme.go` is really cool btw!

**Comment by superbrothers** [comment]
Thanks for the updates. GitHub Actions is now under maintenance, so I'll check the updates tomorrow.

**Comment by droctothorpe** [comment]
I ran `go mod tidy` and committed. Hopefully that fixes the CI error.

**Review by superbrothers — APPROVED** [review]
Great work! :tada:

**Comment by droctothorpe** [comment]
Woohoo! Thanks, @superbrothers.
